keylime/tpm/tpm_main.py

```diff
@@ -40,11 +41,89 @@ def encryptAIK(uuid: str, ek_tpm: bytes, aik_tpm: bytes) -> Optional[Tuple[bytes
             keyblob = base64.b64encode(credentialblob)
 
         except Exception as e:
-            logger.error("Error encrypting AIK: %s", str(e))
+            logger.error("Error encrypting AIK with EK: %s", str(e))
             logger.exception(e)
             raise
 
         return (keyblob, key)
+    
+    @staticmethod
+    def verify_aik_with_iak(uuid: str, aik_tpm: bytes, 
+            iak_tpm: bytes, iak_attest: bytes, iak_sign: bytes) -> bool:
+        
+        # check UUID in certify matches UUID registering
+        if iak_attest[44:(44+len(uuid))] != bytes(uuid, 'utf-8'):
+            logger.warning("Agent %s AIK verification failed, uuid does not match attest info", uuid)
+            return False
+
+        #check aik in certify matches aik being registered
+        if tpm2_objects.get_tpm2b_public_name(aik_tpm) != iak_attest[107:141].hex():
+            logger.warning(" Agent %s AIK verification failed, name of aik does not match attest info", uuid)
+            return False
+
+        # generate digest of attest info
+        attestd, attestpath = tempfile.mkstemp()
+        with open(attestpath, 'wb') as f:
+            f.write(iak_attest)
+        digestd, digestpath = tempfile.mkstemp()
+        command = [
+            "tpm2_hash",
+            "-C",
+            "e",
+            "-g",
+            "sha256",
```

Comment:
This shouldn't have to be hardcoded. I would think you should be able to find the TPM identifier for the hash that was used for creating the signature in some of the data structures involved here.